sample-extension-plugin/build.gradle

```diff
@@ -34,13 +35,7 @@ dependencies {
 
 testingConventions.enabled = false;
 
-
-integTestCluster {
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
-}
-
-run {
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
+testClusters.integTest {
+    testDistribution = 'oss'
+    // plugin(rootProject.getPath())
```

Comment:
Is this not needed anymore?

Reply:
This line is to install job-scheduler before install sample plugin. Not sure if new ES test framework will install job-scheduler automatically.

Reply:
This breaks sample plugin.

```
> Task :opendistro-job-scheduler-sample-extension:run FAILED
Exec output and error:
| Output for ./bin/elasticsearch-plugin:-> Installing file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Downloading file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Failed installing file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Rolling back file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Rolled back file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| Exception in thread "main" java.lang.IllegalArgumentException: Missing plugin [opendistro-job-scheduler], dependency of [opendistro-job-scheduler-sample-extension]
|       at org.elasticsearch.plugins.PluginsService.addSortedBundle(PluginsService.java:452)
|       at org.elasticsearch.plugins.PluginsService.sortBundles(PluginsService.java:424)
|       at org.elasticsearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:847)
|       at org.elasticsearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:821)
|       at org.elasticsearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:866)
|       at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:254)
|       at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:224)
|       at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
|       at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125)
|       at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:91)
|       at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125)
|       at org.elasticsearch.cli.Command.main(Command.java:90)
|       at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:47)

FAILURE: Build failed with an exception.
```

Reply:
I found old `plugin` method is in **clusterFormation** which is not used in our new ES test framework. The new `plugin` method in **testCluster** needs a plugin.zip file, so we will need to build or assemble job-scheduler first and then use the plugin.zip for sample plugin tasks.